test/Serilog.Settings.Configuration.Tests/ConfigurationSettingsTests.cs

```diff
@@ -119,6 +119,87 @@ public void ConfigurationAssembliesFromDllScanning()
         Assert.Single(DummyConsoleSink.Emitted);
     }
 
+    [Fact]
+    public void ConfigurationAssembliesFromDllScanningWithInternalMethodInPublicClass()
+    {
+        var json = """
+            {
+                "Serilog": {
+                    "Using": ["TestDummies"],
+                    "WriteTo": ["DummyConsoleInternal"]
+                }
+            }
+            """;
+
+        var builder = new ConfigurationBuilder().AddJsonString(json);
+        var config = builder.Build();
+        var log = new LoggerConfiguration()
+            .ReadFrom.Configuration(
+                configuration: config,
+                readerOptions: new ConfigurationReaderOptions(ConfigurationAssemblySource.AlwaysScanDllFiles) { AllowInternalMethods = true })
```

Comment:
I was confused about the DLL scanning part. I'd just use `new ConfigurationReaderOptions { AllowInternal…` and remove `FromDllScanning` from the test name for the 3 tests that were added.

Reply:
ok